include/vinecopulib/vinecop/implementation/class.ipp

```diff
@@ -340,20 +367,7 @@ inline void
 Vinecop::select_families(Eigen::MatrixXd data,
                          const FitControlsVinecop& controls)
 {
-  check_data(data);
-  data = collapse_data(data);
-
-  if (vine_struct_.get_trunc_lvl() > 0) {
-    tools_select::FamilySelector selector(
-      data, vine_struct_, controls, var_types_);
-    if (controls.needs_sparse_select()) {
-      selector.sparse_select_all_trees(data);
-      vine_struct_ = selector.get_rvine_structure(); // can be truncated
-    } else {
-      selector.select_all_trees(data);
-    }
-    finalize_fit(selector);
-  }
+  select(data, controls);
 }
```

Comment:
and here maybe `controls.set_trunc_lvl(vine_struct_.get_trunc_lvl())` to maintain the old behavior.